client/jetpack-cloud/sections/partner-portal/primary/payment-method-add/index.tsx

```diff
@@ -104,9 +116,15 @@ function PaymentMethodAdd(): ReactElement {
 				} )
 			);
 		} else {
-			page( '/partner-portal/payment-methods/' );
+			page( partnerPortalBasePath( '/payment-methods' ) );
+		}
+	}, [ page, product, issueLicense ] );
+
+	useEffect( () => {
+		if ( product && paymentMethodRequired ) {
```

Comment:
Fixed the conditionals with the suggested changes, the redirect now works as expected.

Reply:
The process is a bit janky, we should make a separate task to make that smoother, which will involve less code and objective PRs for specific improvements.

Reply:
Added a success notice for when the user successfully issues a license.

Reply:
I realized that we would need to test both `returnQueryArg` or `product`, as there's the case when a user has issued a license previously, had removed the payment method or it got expired, and will resume the process clicking on `Assign License` on the licenses list. This would send the user to the `payment-methods/add` page with a `return` query parameter, on that case we should not send the user to `/partner-portal/payment-methods` but send him to the license assignment (which is on the `return` query argument). Similar to what we have for rendering the progress bar.

So the approach would be:

```javascript
	const successCallback = useCallback( () => {
		if ( returnQueryArg || product ) {
			reduxDispatch(
				fetchStoredCards( {
					startingAfter: '',
					endingBefore: '',
				} )
			);
		} else {
			page( partnerPortalBasePath( '/payment-methods' ) );
		}
	}, [ page, product, reduxDispatch, partnerPortalBasePath, returnQueryArg ] );
```